packages/connectivity/src/scp-cf/xsuaa-service.ts

```diff
@@ -7,30 +7,29 @@ import {
 } from './environment-accessor/environment-accessor-types';
 import { ClientCredentialsResponse } from './xsuaa-service-types';
 import { getXsuaaService, resolveServiceBinding } from './environment-accessor';
-import { getIssuerSubdomain } from './subdomain-replacer';
-import { decodeJwt, tenantId } from './jwt';
+import { decodeJwt, getSubdomain, tenantId } from './jwt';
 
 interface XsuaaParameters {
-  subdomain: string | null;
-  zoneId: string | null;
+  subdomain: string | undefined;
+  zoneId: string | undefined;
   serviceCredentials: ServiceCredentials;
   userJwt?: string;
 }
 
 /**
  * Make a client credentials request against the XSUAA service.
  * @param service - Service as it is defined in the environment variable.
- * @param userJwt - User JWT.
+ * @param jwt - User JWT or object containing the `iss` property.
  * @returns Client credentials token.
  */
 export async function getClientCredentialsToken(
   service: string | Service,
-  userJwt?: string | JwtPayload
+  jwt?: string | JwtPayload
 ): Promise<ClientCredentialsResponse> {
-  const jwt = userJwt ? decodeJwt(userJwt) : {};
+  const decodedJwt = jwt ? decodeJwt(jwt) : {};
```

Comment:
I think this line is redundant, as it is already done in both getSubdomain and tenantId

Reply:
True, but then it is done only once instead.

Reply:
Isn't it done multiple times anyway?
`getSubdomain` and `tenantId` do only check if a jwt exists, not if it is already decoded, and will then execute `decodeJwt` regardless, or am I missing something?
(I didn't see how exactly the underlying `decode` function works)

Reply:
Looking further into the decode function, it is indeed only called once :)